### DISCUSSION TITLE
How to send Avalonia logs and custom logs to a single file

### DISCUSSION THREAD

**Discussion by cristinathoughtpennies** [AvaloniaUI/Avalonia#10082] [Q&A] [answered]
Hello!
I am trying to figure out how to send Avalonia logs (i.e. things like Binding errors etc) AND custom logs from my classes into a single file. Could someone point me in the right direction on how to set this up?

**Comment by maxkatz6** [comment, marked as answer]
You can use libraries Serilog, NLog or Microsoft.Extensions.Logging to setup how logs are supposed to work in your app. You can configure it  to be written in the single file, or multiple files, or both and add console logging on top.

Then, having logging set up in the app, you can redirect Avalonia logs to your logging provider. Implement ILogSink interface and set its instance here https://github.com/AvaloniaUI/Avalonia/blob/0.10.18/src/Avalonia.Base/Logging/Logger.cs#L11

**Reply by AlexeyTliss** [reply]
Can we have description and example code of this feature [in docs](https://docs.avaloniaui.net/docs/guides/implementation-guides/logging-errors-and-warnings) so we don't have to google for this issue? :)

**Reply by stevemonaco** [reply]
@AlexeyTliss Open an issue over at https://github.com/AvaloniaUI/avalonia-docs for this.
